File: dxc-forum-ui/src/Services/MentoringService.js
import httpClient from "./BaseURL";

function getAll() {
  return httpClient.get("/mentoring");
}

function AddMentoring(mentor) {
  return httpClient.post("/mentoring", mentor);
}

function getById(id) {
  return httpClient.get(`/mentoring/${id}`);
}

function update(mentoring, id) {
  return httpClient.put(`/mentoring/${id}`, mentoring);
}

function deleteById(id) {
  return httpClient.delete(`/mentoring/${id}`);
}

function totalEmployee() {
  return httpClient.get("/total_employee");
}

export default {
  getAll,
  AddMentoring,
  getById,
  update,
  deleteById,
  totalEmployee,
};

File: dxc-forum/src/main/java/com/dxc/forum/service/MentoringService.java
package com.dxc.forum.service;

import java.util.List;

import org.springframework.web.multipart.MultipartFile;

import com.dxc.forum.entity.Mentoring;

public interface MentoringService {

	
	//Add Data Service
	public Mentoring addMentoring(Mentoring mentoring);
	
	//Get Data Service
	public List<Mentoring> getAllMentoring();
	
	//Get Specific Data using Id Service
	public Mentoring getMentoring(Long id);
	
	//Update Specific Data using Id Service
	public Mentoring updateMentoring(Mentoring mentoring, Long id);
	
	//Delete Specific Data using ID Service
	public void deleteMentoring(Long id);
	
	//Upload Excel Sheet
	public void uploadExcel(MultipartFile file);
	
	public Long totalEmployee();
}
